contracts/Pod.sol

```diff
@@ -68,79 +81,121 @@ contract Pod is Initializable, ReentrancyGuardUpgradeSafe, ERC777UpgradeSafe, Ba
         return unlockTimestamp[_user];
     }
 
-    function deposit(uint256 _amount) external nonReentrant returns (uint256) {
+    function mintShares(address _receiver, uint256 _amount) external nonReentrant returns (uint256) {
+        // Buy Tickets for caller
         address _sender = _msgSender();
-        Ticket _ticket = ticket();
-        IERC20 _assetToken = yieldService().token();
+        _buyTickets(_sender, _amount);
 
-        // Collect collateral from caller
-        _assetToken.transferFrom(_sender, address(this), _amount);
-
-        // Calculate 
+        // Calculate & Mint Pod-Shares
         uint256 _shares = FixedPoint.divideUintByMantissa(_amount, exchangeRateMantissa());
```

Comment:
Minting tickets should not affect the exchange rate.  This means that the shares need to be calculated before tickets are minted, otherwise they will change the exchange rate.

On that note, this function should be tested more heavily to assert the math is correct. 

Something like:

1. Have "Bob" buy tickets
2. Have the Pod win
3. Have "Alice" buy tickets"

Now assert that Alice's underlying balance matches her deposit